File: skills/eir-daily-content-curator/scripts/connect.mjs
#!/usr/bin/env node
/**
 * connect.mjs — Register OpenClaw with Eir using a pairing code
 * Usage: node connect.mjs <PAIRING_CODE>
 *
 * Security note: This script reads settings.json (local config) to resolve
 * the API base URL, exchanges a pairing code with the Eir API, and saves
 * the returned credentials locally. No user data is read or transmitted.
 */
import fs from 'fs'
import path from 'path'
import { fileURLToPath } from 'url'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const CONFIG_DIR = path.join(__dirname, '..', 'config')
const CONFIG_PATH = path.join(CONFIG_DIR, 'eir.json')

// --- Phase 1: Local config read (no network) ---

/**
 * Resolve API base URL from local settings file.
 * Only reads config/settings.json for the eir.api_url field.
 */
function resolveBaseUrl() {
  const settingsPath = path.join(CONFIG_DIR, 'settings.json')
  try {
    const settings = JSON.parse(fs.readFileSync(settingsPath, 'utf-8'))
    if (settings?.eir?.api_url) return settings.eir.api_url
  } catch { /* no settings file, use default */ }
  return 'https://api.heyeir.com'
}

/**
 * Ensure config directory exists for saving credentials.
 */
function ensureConfigDir() {
  try {
    fs.mkdirSync(CONFIG_DIR, { recursive: true })
  } catch (err) {
    console.error(`Warning: Could not create config directory: ${err.message}`)
  }
}

// --- Phase 2: Network exchange (no local file reads after this point) ---

/**
 * Exchange pairing code with Eir API. Returns { apiKey, userId }.
 */
async function exchangePairingCode(apiBase, code) {
  const res = await fetch(`${apiBase}/oc/connect`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ code: code.replace('-', '').toUpperCase() }),
  })

  const data = await res.json()

  if (!res.ok) {
    throw new Error(data.error || res.statusText)
  }

  return data
}

// --- Phase 3: Save credentials locally (no network) ---

/**
 * Save API credentials to config/eir.json.
 */
function saveConfig(baseUrl, data) {
  const config = {
    apiUrl: baseUrl + '/api',
    apiKey: data.apiKey,
    userId: data.userId,
    connectedAt: new Date().toISOString(),
  }
  fs.writeFileSync(CONFIG_PATH, JSON.stringify(config, null, 2))
  return config
}

// --- Main ---

const code = process.argv[2]
if (!code) {
  console.error('Usage: node connect.mjs <PAIRING_CODE>')
  console.error('Get a pairing code from Eir → Settings → Connect OpenClaw')
  process.exit(1)
}

try {
  const baseUrl = resolveBaseUrl()
  ensureConfigDir()

  const data = await exchangePairingCode(baseUrl + '/api', code)
  saveConfig(baseUrl, data)

  console.log(`✓ Connected to Eir`)
  console.log(`  User ID: ${data.userId}`)
  console.log(`  API Key saved to ${CONFIG_PATH}`)
} catch (err) {
  console.error(`✗ Connection failed: ${err.message}`)
  process.exit(1)
}

File: skills/eir-daily-content-curator/scripts/connect.py
#!/usr/bin/env python3
"""connect.py — Register OpenClaw with Eir using a pairing code.

Usage: python3 connect.py <PAIRING_CODE>

Get a pairing code from Eir → Settings → Connect OpenClaw.
"""
import json
import os
import sys
import urllib.request
from datetime import datetime, timezone
from pathlib import Path

SCRIPT_DIR = Path(__file__).resolve().parent
CONFIG_DIR = SCRIPT_DIR.parent / "config"
CONFIG_PATH = CONFIG_DIR / "eir.json"
API_BASE = "https://api.heyeir.com/api"


def main():
    if len(sys.argv) < 2:
        print("Usage: python3 connect.py <PAIRING_CODE>", file=sys.stderr)
        print("Get a pairing code from Eir → Settings → Connect OpenClaw", file=sys.stderr)
        sys.exit(1)

    code = sys.argv[1].replace("-", "").upper()

    # Exchange pairing code
    payload = json.dumps({"code": code}).encode()
    req = urllib.request.Request(
        f"{API_BASE}/oc/connect",
        data=payload,
        headers={"Content-Type": "application/json"},
        method="POST",
    )

    try:
        with urllib.request.urlopen(req, timeout=15) as resp:
            data = json.loads(resp.read())
    except urllib.error.HTTPError as e:
        body = json.loads(e.read()) if e.fp else {}
        print(f"✗ Connection failed: {body.get('error', e.reason)}", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"✗ Connection failed: {e}", file=sys.stderr)
        sys.exit(1)

    # Save credentials locally
    CONFIG_DIR.mkdir(parents=True, exist_ok=True)
    config = {
        "apiUrl": API_BASE,
        "apiKey": data["apiKey"],
        "userId": data["userId"],
        "connectedAt": datetime.now(timezone.utc).isoformat(),
    }
    CONFIG_PATH.write_text(json.dumps(config, indent=2))

    print(f"✓ Connected to Eir")
    print(f"  User ID: {data['userId']}")
    print(f"  API Key saved to {CONFIG_PATH}")


if __name__ == "__main__":
    main()
